### DISCUSSION TITLE
How to restore data in table if i run destractive query?

### DISCUSSION THREAD

**Discussion by Night-coyote** [supabase/supabase#29221] [Questions] [answered]
I run destractive query that clear tables that i dont expect. I loose information. Is there any ways to restore it on free tier? Or if i swith to paid plan after this disaster. Please help

TRUNCATE TABLE public.locations RESTART IDENTITY CASCADE;  i dont need this cascade(

ALTER SEQUENCE public.locations_location_id_seq RESTART WITH 1;

**Comment by GaryAustin1** [comment, marked as answer]
You have to have backups to restore from.  You can do your own back ups with pg_dump on Free plan, or they are provided on pro plans.

There is no reverse operation in Postgres.
